src/shared/services/app-globals.service.ts

```diff
@@ -22,27 +22,66 @@
 
 import { Injectable, TemplateRef } from '@angular/core';
 import { ReplaySubject } from 'rxjs/ReplaySubject';
-import { SchemaValidationErrors } from '../interfaces';
+import { CategorizedValidationErrors, SchemaValidationErrors } from '../interfaces';
+import { ErrorMapUtilService } from './error-map-util.service';
 
 @Injectable()
 export class AppGlobalsService {
-  private _globalErrorsSubject: ReplaySubject<SchemaValidationErrors> = new ReplaySubject<any>(1);
+  private _externalCategorizedErrorsSubject = new ReplaySubject<CategorizedValidationErrors>(1);
+  private _internalCategorizedErrorsSubject = new ReplaySubject<CategorizedValidationErrors>(1);
+  private _externalErrorCountersSubject = new ReplaySubject<{errors: number, warnings: number}>(1);
+  private _internalErrorCountersSubject = new ReplaySubject<{errors: number, warnings: number}>(1);
+  private internalErrorMap: SchemaValidationErrors = {};
   public adminMode = false;
   public activeTabName = '';
   public tabNameToFirstTopLevelElement: { [tabName: string]: string } = {};
   public templates: { [templateName: string]: TemplateRef<any> };
+  public internalCategorizedErrorMap: CategorizedValidationErrors = { Errors: {}, Warnings: {} };
+  public externalCategorizedErrorMap: CategorizedValidationErrors = { Errors: {}, Warnings: {} };
 
-  set globalErrors(errors: SchemaValidationErrors) {
-    this._globalErrorsSubject.next(errors);
+  constructor(public errorMapUtilService: ErrorMapUtilService) { }
+
+  get externalCategorizedErrorsSubject(): ReplaySubject<CategorizedValidationErrors> {
+    return this._externalCategorizedErrorsSubject;
+  }
+
+  get externalErrorCountersSubject(): ReplaySubject<{errors: number, warnings: number}> {
+    return this._externalErrorCountersSubject;
+  }
+
+  get internalCategorizedErrorsSubject(): ReplaySubject<CategorizedValidationErrors> {
+    return this._internalCategorizedErrorsSubject;
+  }
+
+  get internalErrorCountersSubject(): ReplaySubject<{errors: number, warnings: number}> {
+    return this._internalErrorCountersSubject;
+  }
+
+  set externalErrors(errors: SchemaValidationErrors) {
+    let {categorizedErrorMap, errorCounter, warningCounter} = this.errorMapUtilService.categorizeErrorMap(errors);
+    this.externalCategorizedErrorMap = categorizedErrorMap;
+
+    this.externalCategorizedErrorsSubject.next(this.externalCategorizedErrorMap);
+    this.externalErrorCountersSubject.next({
+      errors: errorCounter,
+      warnings: warningCounter
+    });
   }
 
-  get globalErrorsSubject(): ReplaySubject<SchemaValidationErrors> {
-    return this._globalErrorsSubject;
+  extendInternalErrors(errors: SchemaValidationErrors) {
```

Comment:
instead of abusing `SchemaValidationErrors` we could do `(path: string, errors: Array<{type: string, message: string}>)` (you could also create an interface for the array element type)